src/assets/map.tsx

```diff
@@ -0,0 +1,30 @@
+// Import Mapbox GL, and CSS
+import { useEffect, useRef } from 'react';
+import mapboxgl from 'mapbox-gl';
+import 'mapbox-gl/dist/mapbox-gl.css';
+
+// Set Mapbox access token
+mapboxgl.accessToken = import.meta.env.VITE_MAP_BOX_KEY;
+
+const Map = () => {
+  const map = useRef<mapboxgl.Map>();
```

Comment:
I'm not sure if two refs are needed since it was working fine with just one, but I took a look at mapboxgl's example and it uses two
https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/

Reply:
Spent last 5min trying to understand the two refs and I came away without much except that this may pay off down the line when we add more layers to the map. I agree - might as well follow the tutorial.